### PR TITLE
adding withCheck directive

### PR DESCRIPTION
**Pull request by fr0stbyte**
A new method _withCheck(CheckResult)_ has been added. This method takes in as parameter a function from com.datastax.driver.core.ResultSet => Option[String]. The function should return a String describing the error of the check, None if no error has been encountered.

The list of checks is added to CqlAttributes, and it is executed against the resultSet as obtained in the CqlResponseHandler. If multiple checks fail, their results are all reported.

No tests have been added at this time, but I have tested it in a scenario forcing the failure.

### CODE REVIEW

**Comment by slandelle** [comment]
The proper abstraction for this in Gatling is Validation, not Option.

**Comment by fr0stbyte** [comment]
@slandelle thanks for the feedback, updated to use Validation instead

File: src/main/scala/io/github/gatling/cql/package.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package io.github.gatling
+
+import io.gatling.core.validation.Validation
+import com.datastax.driver.core.ResultSet
+
+package object cql {
+  type CheckResult = ResultSet => Validation[Boolean]
```
Comment (slandelle) [inline src/main/scala/io/github/gatling/cql/package.scala]: This looks more like a CqlCheck than a CheckResult (which would be the resulting validation) :)

**Comment by fr0stbyte** [comment]
totally valid comment, I like CqlCheck better too

**Comment by Mishail** [comment]
thanks guys! I'll take a look

File: src/main/scala/io/github/gatling/cql/CqlRequestBuilder.scala [hunk 2]
```diff
@@ -41,5 +42,6 @@ case class CqlRequestParamsBuilder(tag: String, prepared: PreparedStatement) {
 
 case class CqlRequestBuilder(attr: CqlAttributes) {
     def consistencyLevel(level: ConsistencyLevel) = CqlRequestBuilder(attr.copy(cl= level))
+    def withCheck(check: CqlCheck) = CqlRequestBuilder(attr.copy(checks = check :: attr.checks))
```
Comment (slandelle) [inline src/main/scala/io/github/gatling/cql/CqlRequestBuilder.scala]: Why not just `check()` so it's consistent with other protocol supports?

Comment (fr0stbyte) [inline src/main/scala/io/github/gatling/cql/CqlRequestBuilder.scala]: I was mimicking it off the ```withParams```, but certainly ```check``` makes sense to me

File: src/main/scala/io/github/gatling/cql/CqlResponseHandler.scala [hunk 3]
```diff
@@ -31,18 +31,32 @@ import io.gatling.core.result.message._
...
   with StrictLogging {
-  
+
   private def writeData(status: Status, message: Option[String]) = writeRequestData(session, tag, start, nowMillis, session.startDate, nowMillis, status, message, Nil)
-  
+
   def onSuccess(result: ResultSet) = {
-    writeData(OK, None)
-    next ! session.markAsSucceeded
+    val checkResults: List[String] = checks.flatMap { f => f(result) match {
```
Comment (Mishail) [inline src/main/scala/io/github/gatling/cql/CqlResponseHandler.scala]: I think `check` is more readable than `f` here.  And we should emphasize that we only care about `Failures` here
```scala    
    val checkResults: List[String] = checks.flatMap { check =>
      check(result) match {
        case Failure(msg) => Some(msg)
        case _            => None
      }
    }
```

File: src/main/scala/io/github/gatling/cql/CqlResponseHandler.scala [hunk 4]
```diff
@@ -31,18 +31,33 @@ import io.gatling.core.result.message._
...
   def onSuccess(result: ResultSet) = {
-    writeData(OK, None)
-    next ! session.markAsSucceeded
+    val checkResults: List[String] = checks.flatMap { check =>
+      check(result) match {
+        case Failure(msg) => Some(msg)
+        case _ => None
+      }
+    }
+    checkResults match {
```
Comment (Mishail) [inline src/main/scala/io/github/gatling/cql/CqlResponseHandler.scala]: IMO, `match` is a little bit overkill here. Can we use just `if`?
```scala
    if (checkFailures.isEmpty) {
      writeData(OK, None)
      next ! session.markAsSucceeded
    } else {
      val errors = checkFailures.mkString("\n")
      logger.error(errors)
      writeData(KO, Some(s"Error verifying results: $errors"))
      next ! session.markAsFailed
    }
```

File: src/main/scala/io/github/gatling/cql/CqlResponseHandler.scala [hunk 5]
```diff
@@ -31,18 +31,33 @@ import io.gatling.core.result.message._
...
   with StrictLogging {
-  
+
   private def writeData(status: Status, message: Option[String]) = writeRequestData(session, tag, start, nowMillis, session.startDate, nowMillis, status, message, Nil)
-  
+
   def onSuccess(result: ResultSet) = {
-    writeData(OK, None)
-    next ! session.markAsSucceeded
+    val checkResults: List[String] = checks.flatMap { check =>
```
Comment (Mishail) [inline src/main/scala/io/github/gatling/cql/CqlResponseHandler.scala]: I believe `checkFailures` is a better name for that

File: src/main/scala/io/github/gatling/cql/CqlResponseHandler.scala [hunk 6]
```diff
@@ -31,18 +31,33 @@ import io.gatling.core.result.message._
...
+      check(result) match {
+        case Failure(msg) => Some(msg)
+        case _ => None
+      }
+    }
+    checkResults match {
+      case Nil =>
```
Comment (Mishail) [inline src/main/scala/io/github/gatling/cql/CqlResponseHandler.scala]: IMO, `match` is a little bit overkill here. Can we use just `if`?
```scala
    if (checkFailures.isEmpty) {
      writeData(OK, None)
      next ! session.markAsSucceeded
    } else {
      val errors = checkFailures.mkString("\n")
      logger.error(errors)
      writeData(KO, Some(s"Error verifying results: $errors"))
      next ! session.markAsFailed
    }
```

**Comment by Mishail** [comment]
@fr0stbyte could you please add a sample call to `check` to `CqlCompileTest` as well?

**Comment by fr0stbyte** [comment]
@Mishail @slandelle updated

**Comment by Mishail** [comment]
@fr0stbyte thanks!

**Pull request merged**